### PR TITLE
context cmd cleanup

### PR DESCRIPTION
**Pull request by ransomw1c**
motivated by #308

### CODE REVIEW

File: cmd/datamon/cmd/root.go [hunk 1]
```diff
@@ -110,7 +110,8 @@ func initConfig() {
 	// `viper.ConfigFileUsed()` returns path to config file if error is nil
 	var err error
 	// 2. Initialize config
-	config, err = newConfig()
+	config := new(CLIConfig)
```
Comment (fredbi) [inline cmd/datamon/cmd/root.go]: I believe we should now be able to remove the global `config` var, line 45

File: cmd/datamon/cmd/config.go [hunk 2]
```diff
@@ -19,31 +21,23 @@ type CLIConfig struct {
...
-	return &config, nil
-}
-
-func (c *CLIConfig) setDatamonParams(flags *flagsT) {
-	if flags.context.Descriptor.Name == "" {
-		flags.context.Descriptor.Name = c.Context
-	}
+func populateRemoteConfig() {
+	var flags *flagsT
+	flags = &datamonFlags
```
Comment (fredbi) [inline cmd/datamon/cmd/config.go]:
```
flags := &datamonFlags
```

File: cmd/datamon/cmd/context_create.go [hunk 3]
```diff
@@ -27,9 +27,11 @@ func createContext() {
 	if err != nil {
```
Comment (fredbi) [inline cmd/datamon/cmd/context_create.go]: I am assuming we should populate config here too (cf. panic in unit testing)

File: cmd/datamon/cmd/config.go (hunk 2 above)
Comment (ransomw1c) [reply cmd/datamon/cmd/config.go]: this is an ongoing stylistic discussion:  whether to hindley-milner or not to hindley-milner.

from previous conversations with ritesh, i've variously considered type inference super and harmful.

i do not have a thorughgoing, apply-everywhere stance, except i do recall that i wanted explicit declaration by type separate from assignment here for some reason.  i think simple b/c i found it difficult to remember, in whatever context of a mental model i have for the `cmd/` package environment what with the rapid proliferation of global non-function variables, the type assumptions on `datamonFlags` at this particular part of the codebase, and i actually wanted the reminder.

**Comment by ransomw1c** [comment]
this is a draft.  it's open for review yet not in the same state of readiness for review as, say, #314 .

File: cmd/datamon/cmd/config.go (hunk 2 above)
Comment (fredbi) [reply cmd/datamon/cmd/config.go]: ok. Use // nolint hint then

**Comment by ransomw1c** [comment]
closing this down for now.. please do not delete branch

**Pull request closed without merging**